library/os_manila_share.py

```diff
@@ -1,122 +1,65 @@
         ),
         supports_check_mode=False
     )
 
-    share_client = get_share_client(module)
-
+    result = dict(changed=False)
     share_name = module.params['name']
-    share = get_share_details(share_client, share_name)
-
     user = module.params['user']
-    share_id = share['id']
-    access_key = get_access_key(share_client, share_id, user)
-
-    if not access_key:
-        # TODO we could add access here
-        module.fail_json(msg="User does not have access to share.")
-    share['access_key'] = access_key
-
-    required_protocal = module.params.get('protocol')
-    if required_protocal and share['protocol'] != required_protocal:
-        module.fail_json(
-            msg="Protocol %s does not match requested." % share['protocol'])
-
-    required_size = module.params.get('size')
-    if required_size and share['size'] != required_size:
-        # TODO if size doesn't match, we could expand the share
-        module.fail_json(
-            msg="Size %s does not match requested." % share['size'])
-
-    module.exit_json(changed=False, details=share)
+    conn = openstack.connection.from_config()
+    share = conn.share.find_share(share_name)
+    if share is None:
+        module.fail_json('No share named %r found' % share_name)
+    
+    # the share object doesn't actually expose lots of stuff from the API, including the exports
+    share_details = conn.share.get(f"/shares/{share.id}").json()['share']
+    # module.fail_json(share_details) # DEBUG
+    protocol = share_details['share_proto'].upper()
+    if protocol != 'CEPHFS':
+        module.fail_json("Share named %r is protocol %r, can only handle CEPHFS" % (share_name, protocol))
+    
+    # find host and path from CephFS export of form 'mon1:port,mon2:port,mon3:port:path'
+    host, export_path = share_details['export_location'].rsplit(':', 1)
+
+    # find rule:
+    access_rules = list(conn.share.access_rules(share))
+    if len(access_rules) == 0:
+        module.fail_json("No rules for share named %r" % share_name)
+    if user is None:
+        if len(access_rules) == 1:
+            rule = access_rules[0]
+        else:
+            module.fail_json("Multiple rules found for share named %r but no user specified" % share_name)
+    else:
+        rules = [v for v in access_rules if v.access_to == user]
+        if len(rules) == 0:
+            module.fail_json("No rules found with 'access_to=%s' for share named %r" % (access_to, share_name))
+        if len(rules) != 1:
+            module.fail_json("Multiple rules found with 'access_to=%s' for share named %r" % (access_to, share_name))
+        rule = rules[0]
+
+    # Put required parameters at top level in result to make loops easy:
+    result['host'] = host
+    result['export'] = export_path
+    result['access_key'] = rule['access_key']
+    result['share_user'] = rule['access_to'] # supports case where user is autodetected
+    result['share_name'] = share_name
+    
+    # TODO: add more info here?
```

Comment:
NIT: lets remove this. I would be tempted to add the read only vs read/write info from the access rule.